components/core/file_sender/sender.py

```diff
@@ -0,0 +1,24 @@
+from Server import *
+from flask import Response
+from Main import logging
+
+
+@server.route("/api/get/file", methods=['GET'])
+def send_file_from_path():
+    try:
+        gid = request.args.get('gid')
+        logging.info('User asked for file, having GID ' + gid)
+        id = get_id_from_gid(gid)
+        download_path = get_download_path(id)
+        download_name = get_download_name_from_id(id)
+    except Exception as e:
+        logging.error(" get file API (/api/get/file) got wrong arguments, thrown an error :: %s" % e)
+        return Response("error", status=200)
+
+    try:
+        return send_file(filename_or_fp=download_path, attachment_filename=download_name,
+                         as_attachment=True)
+    except Exception as e:
+        logging.error("File sending has a exception. We got :: %s on file sending" % e)
```

Comment:
I think you have missed this error to get to the end.

Reply:
I think it's correct? Are you saying that this error should be placed after last return?
Actually error returns some text in response so it's that return statement

Reply:
No, this is relevant to my previous comment about placing the error object in the log message. 

```
logging.error("File sending has a exception. When sending file, we got :: %s" % e)
```

Reply:
oh, sorry I got it wrong. So sorry, ok let me change that